x/merkledb/trieview.go

```diff
@@ -509,18 +507,20 @@ func (t *trieView) commitChanges(ctx context.Context, trieToCommit *trieView) er
 	defer span.End()
 
 	switch {
-	case t.isInvalid():
+	case t.invalidated.Get():
 		// don't apply changes to an invalid view
 		return ErrInvalid
 	case trieToCommit == nil:
 		// no changes to apply
 		return nil
-	case trieToCommit.getParentTrie() != t:
+	case trieToCommit.parentTrie.Get() != t:
 		// trieToCommit needs to be a child of t, otherwise the changes merge would not work
 		return ErrViewIsNotAChild
-	case trieToCommit.isInvalid():
+	case trieToCommit.invalidated.Get():
 		// don't apply changes from an invalid view
 		return ErrInvalid
+	case t.committed:
```

Comment:
If a parent is committed while a child is attempting to commitToParent, this could happen.  If the parent had finished committing before the child ran, the child would have had its parent updated to its parent's parent, so this should be equivalent.